kore/src/Kore/Step/Axiom/UserDefined.hs

```diff
@@ -67,13 +67,15 @@ equalityRuleEvaluator
     -- ^ Map from axiom IDs to axiom evaluators
     -> TermLike variable
     -- ^ The function on which to evaluate the current function.
+    -> Pattern.Predicate variable
```

Comment:
I can't see right now why you ended up having to add this parameter here. I would have assumed you only add these parameters if they are required by some function, but you don't seem to use the argument anywhere.